airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -538,10 +540,15 @@ private void close() {
 
   private boolean isTerminal(final Pod pod) {
     if (pod.getStatus() != null) {
-      return pod.getStatus()
+      // Check if "main" container has terminated, as that defines whether the parent process has
+      // terminated.
+      final ContainerStatus mainContainerStatus = pod.getStatus()
           .getContainerStatuses()
           .stream()
-          .anyMatch(e -> e.getState() != null && e.getState().getTerminated() != null);
+          .filter(containerStatus -> containerStatus.getName().equals("main"))
```

Comment:
Can we make this and `.withName("main")`use a variable so if we rename it we don't miss the usages?